File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
function solution(board, moves) {
  const stack = Array(moves.length).fill(0);
  let top = -1;
  let result = 0;

  moves.forEach((move) => {
    move -= 1;

    for (let i = 0; i < board.length; i++) {
      if (board[i][move]) {
        top += 1;
        stack[top] = board[i][move];
        board[i][move] = 0;

        if (top > 0 && stack[top] === stack[top - 1]) {
          top -= 2;
          result += 2;
        }

        break;
      }
    }
  });

  return result;
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    stack = []
    result = 0

    for move in moves:
        move -= 1

        for i in range(len(board)):
            if board[i][move] != 0:
                stack.append(board[i][move])
                board[i][move] = 0

                if len(stack) > 1 and stack[-1] == stack[-2]:
                    stack.pop()
                    stack.pop()
                    result += 2

                break

    return result
